### ISSUE TITLE
[flutter_tools] allow filtering of adb logcat by "flutter" to be configured from CLI args

### ISSUE THREAD

**Issue by christopherfujino** [flutter/flutter#120711]
ADB logcat logs are very verbose, so we filter it by lines that contain the pattern `'flutter'`. We should make this configurable via a command-line flag, so that developers who want to debug the full logs can see this. This would also be leveraged by editor plugins, since--for example--Android Studio provides a UI for viewing full logcat logs in a natively launched application.

cc @jacob314

**Comment by DanTup** [comment]
If the volume of traffic is high, there may be value in allowing it to be stopped/started while the app is running (via a message to the `flutter run` daemon, or over the a VM Service extension).

This could allow the user to start the log, perform some action in the app, then stop the log, rather than having to wade through large amounts of logging.

**Comment by lukehutch** [comment]
Some logcat lines don't contain the string `flutter` (unless that part of the line has already been suppressed):

```
D/EGL_emulation( 4571): app_time_stats: avg=39.67ms min=0.97ms max=650.57ms count=26
```

This debug line in particular really pollutes the log.

The prefix gives the log level: `D/` is debug, `W/` is warning, `E/` is error. So in addition to the proposal of enabling or disabling the filter for logcat lines containing `flutter`, I suggest also allowing the user to set a minimum loglevel.

**Comment by LoveriusB** [comment]
Any news on this? It really makes it impossible to work with my terminal.

**Comment by matanlurey** [comment]
Fwiw, I looked into this briefly today.

> ADB logcat logs are very verbose, so we filter it by lines that contain the pattern `'flutter'`.

... is not totally true - there are two branches:

- Current app process (i.e. your flutter app): virtually no filtering
- _Another_ process (i.e. another app, possibly Flutter): filtering based on things like `flutter`

The vast majority of so-called log-spam (e.g. `D/EGL_emulation`) are in the _current_ process ID, meaning its not filtered.

I am very supportive to adding much more _liberal_ `adb logcat` filtering, similar to what we did in the engine:

https://github.com/flutter/engine/blob/main/testing/scenario_app/bin/utils/adb_logcat_filtering.dart

... however that is a potentially larger change. If the tool team is supportive of more aggressive Android filtering, I'd be happy to contribute here; removing `triaged-tool` to get an opinion before moving forward.

**Comment by reidbaker** [comment]
yes filtering is safe and personally when developing I would prefer to see level filtering and the exclusion of anything out of process without and opt in. If I want to see what is happening in android more broadly I should use adb the flutter tools should help me debug my app.

**Comment by bkonyi** [comment]
Thanks Reid! I think this is something we should move forward with at some point.
